### DISCUSSION TITLE
Zigbee module and ESP8266

### DISCUSSION THREAD

**Discussion by array81** [ptvoinfo/zigbee-configurable-firmware#214] [General]
I'd like connect a Zigbee module Ti CC2530 development board (https://it.aliexpress.com/item/1005003670231117.html?gatewayAdapt=glo2ita) to Wemos D1 (esp8266) then use the CC2530 only to send and receive data to/from coordinator (by zigbee2mqtt).

ZigBee coordinator -> CC2530 -> Wemos and Wemos -> CC2530 -> ZigBee coordinator

All the logic would be managed by the esp8266, the zigbee module would be used only for passing data.

Is it possible? Suggestions on cabling and firmware setup?

**Comment by ptvoinfo** [comment]
Hi!

What kind of data would you like to send? For example, CC2530 can work as a transparent UART data bridge. But it is not possible to retransmit all Zigbee traffic to Wemos.

**Reply by array81** [reply]
Thanks for reply.

This is my idea:
Home Assistant <-> Zigbee2Mqtt <-> Sonoff Zigbee Dongle (Coordinator) <-> My Device (Wemos + CC2530) <-> Siren
HA send a boolean dato to Wemos (by zigbee no by wifi) then it handles the logic and if necessary turn on/off the Siren.
I already have experience with esp8266/esp32 but no experience with zigbee.

**Comment by ptvoinfo** [comment]
You can add a GPIO output to CC2530 and control it as GPIO input in Wemos.

**Reply by array81** [reply]
I nedd add a GPIO output on PTVO firmware?
Is there a exmaple of calble connection between CC2530 and Wemos? Can I connect only GND, V and this pin?

**Reply by ptvoinfo** [reply]
Yes, you need to configure the firmware to use one GPIO output on any pin. Please note, you should use 3.3V voltage as a source voltage for CC2530. Maybe, you should use a level shifter for GPIO.

**Comment by array81** [comment]
Do you know if it is possible to use CCLoader + ESP8266 to flash firmware without using debugger?

**Comment by tamas2001** [comment]
Yes, I am flashing it on that way!On 2023. 07. 22., 14:09 Martin ***@***.***> wrote:

   
  Do you know if it is possible to use CCLoader + ESP8266 to flash firmware without using debugger?

**Reply by array81** [reply]
good. thanks

**Reply by JonRobert** [reply]
Yes, I've been doing it for a while.   See this link for more details  [https://community.hubitat.com/t/thingshield-replacement-for-arduino-over-zigbee-now-available-for-hubitat/32541/98](url)

I use this batch file for ease in downloading. 

```
C:\utilities\ptvo\Convert-n-download>echo "syntax:  CLD.bat <filename w/o extension>    <com>"
"syntax:  CLD.bat <filename w/o extension>    <com>"

C:\utilities\ptvo\Convert-n-download>objcopy.exe --gap-fill 0xFF --pad-to 0x040000 -I ihex .hex -O binary .bin
objcopy.exe: '.hex': No such file

C:\utilities\ptvo\Convert-n-download>CCLoader.exe  .bin 0
Copyright (c) 2013 RedBearLab.com
CCLoader.exe version 0.5
Invalid parameters.
Usage: CCLoader.exe <com number> <bin file> <device>
Example: CCLoader.exe 2 abc.bin 0
 <device>: 0 -- Default (e.g. UNO)
           1 -- Leonardo

C:\utilities\ptvo\Convert-n-download>pause
Press any key to continue . . .


```





`

**Comment by array81** [comment]
I managed to compile the firmware and upload it to my CC2530 module through an esp8266. The firmware works.
In my intentions the CC2530 module will be connected to Zigbee2Mqtt and therefore to Home Assistant.
I need 4 on/off type PINs. I try to explain myself.
I want from Home Assistant to be able to change the on/off value (as if they were buttons) through which to send commands to the esp8266 connected to my CC2530 module. But I also wish my esp8266 could do the same thing: change the state of the CC2530 pins (and thus the state of the Home Assistant buttons).
I don't understand how to configure PINs on PTVO firmware.
I guess I have to connect the input and output PINs.
Do I have to use virtual PINs?
Or should I use separate input and output PINs and bind all PINs to the esp8266?
